src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final String DATE_PATTERN = "dd.MM.yyy";
+    private static final String EMPTY_SEPARATOR = " ";
+    private static final String REPORT_SEPARATOR = " - ";
+    private static final short LINE_DATE = 0;
+    private static final short LINE_NAME = 1;
+    private static final short LINE_HOURS = 2;
+    private static final short LINE_SALARY = 3;
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_PATTERN);
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder report = new StringBuilder("Report for period ")
+                .append(dateFrom).append(REPORT_SEPARATOR)
+                .append(dateTo).append(System.lineSeparator());
+        for (String name : names) {
+            report.append(name).append(REPORT_SEPARATOR)
+                    .append(calculateSalary(name, data, dateFrom, dateTo))
+                    .append(System.lineSeparator());
+        }
+        return report.toString().trim();
+    }
+
+    private int calculateSalary(String name, String[] data, String fromDate, String toDate) {
+        int salary = 0;
+        String[] table;
```

Comment:
You can remove this variable declaration here and do it directly on line 32. Also it should have more clear naming